server/http_test.go

```diff
@@ -842,3 +955,477 @@ func TestGetSubscriptionsForChannel(t *testing.T) {
+			apiCalls:             hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Initial Subscription Template, long name provided": {
+			subscriptionTemplate: `{"instance_id": "jiraurl1", "name": "` + TestDataLongSubscriptionName + `", "channel_id": "mockChannelID_____________", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": ["10001"]}}`,
+			expectedStatusCode:   http.StatusInternalServerError,
+			apiCalls:             hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Initial Subscription Template, no project provided": {
+			subscriptionTemplate: `{"instance_id": "jiraurl1", "name": "mockName", "channel_id": "mockChannelID_____________", "filters": {"events": ["jira:issue_created"], "projects": [], "issue_types": ["10001"]}}`,
+			expectedStatusCode:   http.StatusInternalServerError,
+			apiCalls:             hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Initial Subscription Template, no issue types provided": {
+			subscriptionTemplate: `{"instance_id": "jiraurl1", "name": "mockName", "channel_id": "mockChannelID_____________", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": []}}`,
+			expectedStatusCode:   http.StatusInternalServerError,
+			apiCalls:             hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Adding to existing templates with other channel": {
+			subscriptionTemplate: `{"instance_id": "jiraurl1", "name": "some name", "channel_id": "mockChannelID_____________", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": ["10001"]}}`,
+			expectedStatusCode:   http.StatusOK,
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{
+				{
+					Filters: getMockSubscriptionFilter("jira:issue_created"),
+				},
+				{
+					Filters: getMockSubscriptionFilter("jira:issue_created"),
+				},
+			},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID:      model.NewId(),
+							Filters: getMockSubscriptionFilter("jira:issue_created"),
+						},
+					}), t),
+		},
+		"Adding to existing templates the in the same channel": {
+			subscriptionTemplate: `{"instance_id": "jiraurl1", "name": "subscription name", "channel_id": "mockChannelID_____________", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": ["10001"]}}`,
+			expectedStatusCode:   http.StatusOK,
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{
+				{
+					Filters: getMockSubscriptionFilter("jira:issue_created"),
+				},
+				{
+					Filters: getMockSubscriptionFilter("jira:issue_updated"),
+				},
+			},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID:      model.NewId(),
+							Filters: getMockSubscriptionFilter("jira:issue_updated"),
+						},
+					}), t),
+		},
+		"Adding to existing templates with same name in the same channel": {
```

Comment:
```suggestion
		"Adding to existing templates with the same name in the same channel": {
```
pls read the comment correctly